vyked/jsonprotocol.py

```diff
@@ -63,28 +82,41 @@ def close(self):
     def data_received(self, byte_data):
         string_data = byte_data.decode()
         self.logger.debug('Data received: %s', string_data)
+
         try:
             pass
             try:
                 string_data = self._partial_data + string_data
                 partial_data = ''
+                json_loads_time = self._json_loads_time
+
                 for e in string_data.split('!<^>!'):
                     if e:
                         try:
+                            start_time = time.time()
                             element = json.loads(partial_data + e)
```

Comment:
One more optimisation we can probably make is skip json.loads in case len(e) == len(string_data). Because this is a case where no delimiter is present in the string. Lets discuss on Monday

Reply:
yup when delimiter is not present the string_data.split('!<^>!') array size will be 1. and in all other cases the split array size will be greater than 1